game/hiscore.cc

```diff
@@ -49,35 +63,62 @@ bool Hiscore::hasHiscore(const SongId& songid) const {
 }
 
 unsigned Hiscore::getHiscore(SongId songid) const {
-	for (auto const& score: m_hiscore) 
-		if (songid == score.songid && currentLevel() == score.level) 
+	for (auto const& score: m_hiscore) {
+		if (songid == score.songid && currentLevel() == score.level) {
 			return score.score;
+		}
+	}
+
 	return 0;
 }
 
+std::vector<HiscoreItem> Hiscore::getHiscores(SongId songid) const {
+	auto scores = std::vector<HiscoreItem>{};
+
+	for (auto const& score: m_hiscore) {
+		if (songid == score.songid && currentLevel() == score.level) {
```

Comment:
Pre filter beforehand to eliminate the `if` and nesting and the forloop in general.
Can be done with a single line `std::copy_if` i assume since we're inserting in a new vector